core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +49,143 @@ public void testAnnouncement() throws InterruptedException {
+        channelMonitorAggregator.start(true, 10000, 1000);
+
+        //need to await for original request for heartbeat to be send to simulate response with same correlationId
+        Message requestMessage = awaitHeartBeatRequestSent(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME);
+
+        Message responseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                payload);
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("HeartBeat Response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void testHeartbeatUnexpectedMessage() throws InterruptedException {
+
+        CountDownLatch heartBeatResponseReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertEquals(0, stats.getTopicInfoMap().size());
+            heartBeatResponseReceived.countDown();
+        };
+
+        ChannelMonitorAggregator channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(true, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, 1000);
+
+        //need to await for original request for heartbeat to be send to simulate response with same correlationId
+        Message requestMessage = awaitHeartBeatRequestSent(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME);
+
+        Message responseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                " unexpected statistics format received");
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("HeartBeat Response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    private Message awaitHeartBeatRequestSent(long timeout) {
+        long startTime = System.currentTimeMillis();
+        String originalRequest = null;
+        while (originalRequest == null && (System.currentTimeMillis() - startTime) < timeout) {
+            originalRequest = MockAdapter.pollJsonMessageForTopic(Utils.TOPIC_HEARTBEAT);
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+        Message requestMessage = Utils.fromJson(originalRequest, Message.class, msbContext.getPayloadMapper());
+        return requestMessage;
+    }
+
+    private Message createMsbRequestMessage(String topicTo, String correlationId, String payloadString) {
```

Comment:
we need to clear methods in TestUtils before adding new there (there are too much there and if ad more it's gonna be total mess ). Think of it as a separate task, Not in scope of this